format/udf-spec.md

```diff
@@ -0,0 +1,221 @@
+
+### Definition-Version
+
+| Requirement | Field name              | Description                                                                                |
+|-------------|-------------------------|--------------------------------------------------------------------------------------------|
+| *required*  | `definition-version-id` | Unique identifier of the definition version, a monotonically increasing long. Example: `2` |
+| *required*  | `timestamp-ms`          | Timestamp when the definition was created or updated. Example: `1734506000456`             |
+| *required*  | `body`                  | List of mapping of overload uuids to their current version ids.                            |
+
+## Function Resolution in Engines
+Resolution rule is decided by engines, but engines SHOULD:
+1. Prefer exact signature matches over casting or subtyping.
+2. Allow numeric widening (e.g., `INT` --> `BIGINT/FLOAT`) by default.
+3. Require explicit casts for risky conversions (e.g., `STRING` --> `DATE`).
+
+## Appendix A: Example
+
+SQL statement:
+
+```sql
+CREATE FUNCTION add_one(x INT COMMENT 'Input integer')
+COMMENT 'Add one to the input value'
+RETURNS INT
+RETURN x + 1;
+
+CREATE FUNCTION add_one(x FLOAT COMMENT 'Input float')
+COMMENT 'Add one to the input value'
+RETURNS FLOAT
+RETURN x + 1.0;
+```
+
+```json
+{
+  "function-uuid": "42fd3f91-bc10-41c1-8a52-92b57dd0a9b2",
+  "format-version": 1,
+  "definition": [
+    {
+      "overload-uuid": "d2c7dfe0-54a3-4d5f-a34d-2e8cfbc34111",
+      "parameters": [
+        { "name": "x", "type": "int", "doc": "Input integer" }
+      ],
+      "return-type": "int",
+      "doc": "Add one to the input integer",
+      "versions": [
+        {
+          "overload-version-id": 1,
+          "deterministic": true,
+          "representations": [
+            {
+              "type": "sql",
+              "dialect": "trino",
+              "body": "x + 2"
+            }
+          ],
+          "timestamp-ms": 1734507000123
+        },
+        {
+          "overload-version-id": 2,
+          "deterministic": true,
+          "representations": [
+            {
+              "type": "sql",
+              "dialect": "trino",
+              "body": "x + 1"
+            }
+          ],
+          "timestamp-ms": 1735507000124
+        }
+      ]
+    },
+    {
+      "overload-uuid": "7c9f93b1-28b4-4ef5-90f5-70c73cda2222",
+      "parameters": [
+        { "name": "x", "type": "float", "doc": "Input float" }
+      ],
+      "return-type": "float",
+      "doc": "Add one to the input float",
+      "versions": [
+        {
+          "overload-version-id": 1,
+          "deterministic": true,
+          "representations": [
+            {
+              "dialect": "trino",
+              "body": "x + 1.0"
+            }
+          ],
+          "timestamp-ms": 1734507001123
+        }
+      ]
+    }
+  ],
+  "definition-versions": [
```

Comment:
I am wondering if we can just add a `current-version` in the overload definition directly, e.g. above line 149?